index.js

```diff
@@ -101,14 +101,18 @@ module.exports = class AsyncValuePromise {
       }
     }
 
-    for (let i = 0; i < count; i++) {
-      promises[i].then(value => {
-        if (failed) return
-        results[i] = value
-        if (--remaining === 0) {
-          promise.resolve(results)
-        }
-      }).catch(fail)
+    if (count === 0) {
+      process.nextTick(() => promise.resolve([]))
```

Comment:
Using `nextTick` aligns with how `Promise.all([]).then(...)` would work, but since we're deviating from this for performance reasons we might not want to align on this here?